### PR TITLE
[ENH] Allow Kernel CD test to pass in custom propensity model and estimates

### LINKED ISSUE #105: Pre-fit propensity score model?
**Is your feature request related to a problem? Please describe.**
If one has already fit a propensity score model, it would be helpful to be able to pass it in.

**Describe the solution you'd like**
some sort of API to provide a pre-fit propensity score model.

### PR DESCRIPTION
**Pull request by adam2392**
Signed-off-by: Adam Li <adam2392@gmail.com>

Fixes #105 

Changes proposed in this pull request:
- Adds two keyword arguments to `KernelCDTest` for propensity model or propensity estimates
- Shares code between `KernelCDTest` and `BregmanCDTest`

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/dodiscover/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/dodiscover/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

**Comment by adam2392** [comment]
@robertness can you give this a quick review? 

Needed for causal-dcorr paper with @ebridge2

File: dodiscover/cd/base.py [hunk 1]
```diff
@@ -52,3 +77,80 @@ def test(
...
+        return self.propensity_est_
+
+    @abstractmethod
+    def _statistic(self, X: ArrayLike, Y: ArrayLike, group_ind: ArrayLike) -> float:
+        """Abstract method for computing the test statistic."""
+        pass
+
+    def compute_null(
+        self, e_hat: ArrayLike, X: ArrayLike, Y: ArrayLike, null_reps: int = 1000, random_state=None
+    ):
```
Comment (jaron-lee) [inline dodiscover/cd/base.py:122]: Missing return type?

File: tests/unit_tests/conditional/cd/test_cd.py [hunk 2]
```diff
@@ -63,48 +67,67 @@ def test_cd_tests_error(cd_estimator):
...
     group_col = "group"
+    alpha = 0.1
 
     if env_type == "single":
         _, pvalue = cd_estimator.test(df, {"x"}, {"x1"}, group_col=group_col)
         print(pvalue)
-        assert pvalue > 0.05
+        assert pvalue > alpha
         _, pvalue = cd_estimator.test(df, {"x"}, {"z"}, group_col=group_col)
         print(pvalue)
```
Comment (jaron-lee) [inline tests/unit_tests/conditional/cd/test_cd.py:119]: Should these prints be left in the test code?

**Review by jaron-lee — DISMISSED** [review]
Overall LGTM, just a few minor nits.

File: tests/unit_tests/conditional/cd/test_cd.py (hunk 2 above)
Comment (adam2392) [reply tests/unit_tests/conditional/cd/test_cd.py:119]: I left those in there to allow easier debugging of which one fails, but I am adding a better fix.

File: dodiscover/cd/base.py (hunk 1 above)
Comment (adam2392) [reply dodiscover/cd/base.py:122]: Added.

**Review by jaron-lee — APPROVED** [review]
LGTM

**Pull request merged**